pkg/operator/controller/ingress/status.go

```diff
@@ -822,25 +819,37 @@ func computeDNSStatus(ic *operatorv1.IngressController, wildcardRecord *iov1.DNS
 			},
 		}
 	}
-
-	if !manageDNSForDomain(ic.Status.Domain, status, dnsConfig) {
-		return []operatorv1.OperatorCondition{
+	var conditions []operatorv1.OperatorCondition
+	if ic.Status.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy == operatorv1.UnmanagedLoadBalancerDNS ||
+		(ic.Spec.EndpointPublishingStrategy != nil && ic.Spec.EndpointPublishingStrategy.LoadBalancer != nil &&
+			ic.Spec.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy == operatorv1.UnmanagedLoadBalancerDNS &&
+			ic.Spec.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy != ic.Status.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy) {
```

Comment:
Does this extra `ic.Spec.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy != ic.Status.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy` provide value?

Isn't this always true because you just checked `ic.Status.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy == operatorv1.UnmanagedLoadBalancerDNS`?